### ISSUE TITLE
Provide a way to query MMap info from stream

### ISSUE THREAD

**Issue by rpattabi** [google/oboe#1032]
We want to know whether the device supports mmap and also whether our stream uses mmap or not. This will help us to diagnose issues with the stream especially for device specific issues. It will also be useful while reporting issues here.

Currently [AudioStream](https://github.com/google/oboe/blob/master/include/oboe/AudioStream.h) interface provides lots of useful information about the stream such as `isXRunCountSupported()`, `getXRunCount()`, etc. We still miss something like `isMMapUsed()` and `isMMapAvailable()` on this interface.

We looked at MMap related code from QuirkManager and OboeTester. It is not trivial to get this information from the stream. A naive attempt calling `QuirksManager::isMMapUsed()` resulted in a crash on our aaudio device which doesn't support mmap. Apparently, calling  `stream_isMMapUsed` on aaudio internally caused this crash (But, we could get `stream_getFramesPerBurst()` working fine with similar code). OTOH, OboeTester does lot more to get this information. We don't know if we should be using similar approach in our production code.

In short, it would be nice if `AudioStream` provides `isMMapUsed()`. Similarly, `isMMapAvailable()` on suitable interface to learn about the device capability would help.
